cmd/govim/flag.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+)
+
+var (
+	flagSet = flag.NewFlagSet("govim", flag.ContinueOnError)
+	fTail   = flagSet.Bool("tail", false, "whether to also log output to stdout")
+)
+
+func init() { flagSet.Usage = usage }
+
+func usage() {
+	fmt.Fprintf(os.Stderr, `
+Usage of govim:
+
+	govim [-tail] /path/to/gopls
+
+`[1:])
+	flagSet.PrintDefaults()
+}
+
+type usageErr string
+
+func (u usageErr) Error() string { return string(u) }
+
+type flagErr string
+
+func (f flagErr) Error() string { return string(f) }
+
+func main() { os.Exit(main1()) }
+
+func main1() int {
+	err := mainerr()
+	if err == nil {
+		return 0
+	}
+	switch err := err.(type) {
+	case usageErr:
```

Comment:
This type should be returned from `mainerr(...)` right? Usage isn't printed as it is now.